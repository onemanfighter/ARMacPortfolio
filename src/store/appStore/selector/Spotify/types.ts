export type SpotifyPlaylistData = {
  name: string;
  url: string;
};
