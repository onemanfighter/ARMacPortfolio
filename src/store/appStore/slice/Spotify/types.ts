export type SpotifyItemData = {
  type: 'artist';
  name: string;
  id: string;
};

export type SpotifyState = {
  state: 'playing' | 'main';
  currentPlaylist: IndianArtist | undefined;
};

export interface SpotifyAppAction {
  /**
   * Open playlist.
   */
  openPlaylist: (playlist: IndianArtist) => void;

  /**
   * Close playlist.
   */
  closePlaylist: () => void;
}

export type SpotifyStateSlice = SpotifyState & SpotifyAppAction;

export enum IndianArtist {
  ARIJIT_SINGH = 'Arijit Singh',
  ATIF_ASLAM = 'Atif Aslam',
  PRITAM = 'Pritam',
  AR_RAHMAN = 'A.R. Rahman',
  SONU_NIGAM = 'Sonu Nigam',
  SHAAN = 'Shaan',
  SHREYA_GHOSHAL = 'Shreya Ghoshal',
  SUNIDHI_CHAUHAN = 'Sunidhi Chauhan',
  KK = 'KK',
  MOHIT_CHAUHAN = 'Mohit Chauhan',
  NEHA_KAKKAR = 'Neha Kakkar',
  TONY_KAKKAR = 'Tony Kakkar',
  BADSHAH = 'Badshah',
  HONEY_SINGH = 'Yo Yo Honey Singh',
  DILJIT_DOSANJH = 'Diljit Dosanjh',
  DARSHAN_RAVAL = 'Darshan Raval',
  SHANKAR_MAHADEVAN = 'Shankar Mahadevan',
  UDIT_NARAYAN = 'Udit Narayan',
}
