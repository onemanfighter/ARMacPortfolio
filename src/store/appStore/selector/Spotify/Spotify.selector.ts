import { AppStoreState } from '../../appStore';

const spotifySelector = (state: AppStoreState) => ({});

export { spotifySelector };
