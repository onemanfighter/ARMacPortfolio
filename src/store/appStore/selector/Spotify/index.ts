export { spotifySelector } from './Spotify.selector';
