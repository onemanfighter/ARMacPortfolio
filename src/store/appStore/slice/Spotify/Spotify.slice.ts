import { AppStoreSlice } from '../../appStore';
import { SpotifyState, SpotifyStateSlice } from './types';

const defaultSpotifyState: SpotifyState = {};

const createSpotifySlice: AppStoreSlice<SpotifyStateSlice> = (set) => ({
  ...defaultSpotifyState,
});

export default createSpotifySlice;
