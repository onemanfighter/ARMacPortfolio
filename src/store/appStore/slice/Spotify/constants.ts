import { IndianArtist, SpotifyItemData } from './types';

export const PLAYLISTS: Record<IndianArtist, SpotifyItemData> = {
  [IndianArtist.ARIJIT_SINGH]: {
    name: 'Arijit Singh',
    type: 'artist',
    id: '4YRxDV8wJFPHPTeXepOstw',
  },
  [IndianArtist.ATIF_ASLAM]: {
    type: 'artist',
    name: 'Atif Aslam',
    id: '2oSONSC9zQ4UonDKnLqksx',
  },
  [IndianArtist.PRITAM]: {
    type: 'artist',
    name: 'Pritam',
    id: '1wRPtKGflJrBx9BmLsSwlU',
  },
  [IndianArtist.AR_RAHMAN]: {
    type: 'artist',
    name: 'A.R. Rahman',
    id: '1mYsTxnqsietFxj1OgoGbG',
  },
  [IndianArtist.SONU_NIGAM]: {
    type: 'artist',
    name: 'Sonu Nigam',
    id: '1dVygo6tRFXC8CSWURQJq2',
  },
  [IndianArtist.SHAAN]: {
    type: 'artist',
    name: 'Shaan',
    id: '5cB4d4jPYjMT326sjihQ4m',
  },
  [IndianArtist.SHREYA_GHOSHAL]: {
    type: 'artist',
    name: 'Shreya Ghoshal',
    id: '0oOet2f43PA68X5RxKobEy',
  },
  [IndianArtist.SUNIDHI_CHAUHAN]: {
    type: 'artist',
    name: 'Sunidhi Chauhan',
    id: '3eDT9fwXKuHWFvgZaaYC5v',
  },
  [IndianArtist.KK]: {
    type: 'artist',
    name: 'KK',
    id: '4fEkbug6kZzzJ8eYX6Kbbp',
  },
  [IndianArtist.MOHIT_CHAUHAN]: {
    type: 'artist',
    name: 'Mohit Chauhan',
    id: '5GnnSrwNCGyfAU4zuIytiS',
  },
  [IndianArtist.NEHA_KAKKAR]: {
    type: 'artist',
    name: 'Neha Kakkar',
    id: '5f4QpKfy7ptCHwTqspnSJI',
  },
  [IndianArtist.TONY_KAKKAR]: {
    type: 'artist',
    name: 'Tony Kakkar',
    id: '0NZtn1Kyq08alpHCTRf3dv',
  },
  [IndianArtist.BADSHAH]: {
    type: 'artist',
    name: 'Badshah',
    id: '0y59o4v8uw5crbN9M3JiL1',
  },
  [IndianArtist.HONEY_SINGH]: {
    type: 'artist',
    name: 'Honey Singh',
    id: '7uIbLdzzSEqnX0Pkrb56cR',
  },
  [IndianArtist.DILJIT_DOSANJH]: {
    type: 'artist',
    name: 'Diljit Dosanjh',
    id: '2FKWNmZWDBZR4dE5KX4plR',
  },
  [IndianArtist.DARSHAN_RAVAL]: {
    type: 'artist',
    name: 'Darshan Raval',
    id: '2GoeZ0qOTt6kjsWW4eA6LS',
  },
  [IndianArtist.SHANKAR_MAHADEVAN]: {
    type: 'artist',
    name: 'Shankar Mahadevan',
    id: '1SJOL9HJ08YOn92lFcYf8a',
  },
  [IndianArtist.UDIT_NARAYAN]: {
    type: 'artist',
    name: 'Udit Narayan',
    id: '70B80Lwx2sxti0M1Ng9e8K',
  },
};
