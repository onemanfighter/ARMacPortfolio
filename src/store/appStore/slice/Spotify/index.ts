export { default as createSpotifySlice } from './Spotify.slice';
export type { SpotifyStateSlice, SpotifyState, SpotifyItemData } from './types';
export { IndianArtist } from './types';
export { PLAYLISTS } from './constants';
