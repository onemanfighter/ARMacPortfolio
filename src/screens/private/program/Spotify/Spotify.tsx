import { IndianArtist, appStore } from '@appStore';
import { SpotifyProps } from './type';
import { useShallow } from 'zustand/react/shallow';
import { spotifySelector } from '@appStore/selector';
import { Box, IconButton } from '@chakra-ui/react';
import { BackIcon } from '@assets';

const Spotify = (props: SpotifyProps) => {
  const { state, getCurrentPlaylist, closePlaylist, openPlaylist } =
    appStore(spotifySelector);

  const a = appStore();

  const currentList = getCurrentPlaylist();

  return (
    <Box width={'100%'} height={'100%'}>
      <Box
        width={'100%'}
        height={8}
        border={state === 'playing' ? '1px solid #1DB954' : '1px solid #000000'}
        display="flex"
        justifyContent={'center'}
        alignItems={'center'}
        color="white"
        columnGap={2}
      >
        <IconButton
          aria-label="back"
          size={'xs'}
          icon={<BackIcon color="white" width={24} height={24} />}
          _active={{
            transform: 'scale(0.9)',
          }}
          colorScheme="white"
          onClick={() => {
            console.log('closePlaylist');
            closePlaylist();
          }}
          // isDisabled={state === 'main'}
        />
        {currentList?.name}
      </Box>

      {state === 'playing' ? (
        <iframe
          title={currentList?.name}
          src={currentList?.url}
          width={'100%'}
          height={'100%'}
          allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture"
          loading="lazy"
        ></iframe>
      ) : (
        <Box></Box>
      )}
    </Box>
  );
};

export default Spotify;
