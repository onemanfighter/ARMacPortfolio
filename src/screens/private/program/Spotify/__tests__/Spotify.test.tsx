import { render } from '@testing-library/react';
import Spotify from '../Spotify';

describe('Spotify', () => {
  it('should render correctly to match snapshot', () => {
    const { container } = render(<Spotify />);

    expect(container).toMatchSnapshot();
  });
});
