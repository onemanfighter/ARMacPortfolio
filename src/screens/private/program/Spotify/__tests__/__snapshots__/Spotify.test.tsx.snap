// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spotify should render correctly to match snapshot 1`] = `
<div>
  <div
    aria-label="spotify"
    class="css-uwwqev"
  >
    <div
      class="css-yecknf"
    >
      <button
        aria-label="back"
        class="chakra-button css-1kxr5pj"
        type="button"
      >
        <svg
          aria-hidden="true"
          color="white"
          fill="currentColor"
          focusable="false"
          height="24"
          viewBox="0 0 512 512"
          width="24"
        >
          <path
            d="M249.38 336L170 256l79.38-80M181.03 256H342"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="32"
          />
          <path
            d="M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z"
            fill="none"
            stroke="currentColor"
            stroke-miterlimit="10"
            stroke-width="32"
          />
        </svg>
      </button>
      A.R. Rahman
    </div>
    <iframe
      allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture"
      height="100%"
      loading="lazy"
      src="https://open.spotify.com/embed/artist/1mYsTxnqsietFxj1OgoGbG?utm_source=generator"
      title="A.R. Rahman"
      width="100%"
    />
  </div>
</div>
`;
